File: reimi/backend/reimi_app/src/main/java/com/reimi/reimi_app/domain/model/weatherpersonality/AnswerChoice.java
package com.reimi.reimi_app.domain.model.weatherpersonality;

public enum AnswerChoice {

    STRONGLY_A(2),
    SLIGHTLY_A(1),
    SLIGHTLY_B(-1),
    STRONGLY_B(-2);

    private final int score;

    AnswerChoice(int score) {
        this.score = score;
    }

    public int score() {
        return score;
    }

    public static AnswerChoice from(String value) {
        try {
            return AnswerChoice.valueOf(value);
        } catch (IllegalArgumentException e) {
            throw new IllegalArgumentException("選択肢以外の回答が含まれています");
        }
    }
}

File: reimi/frontend/reimi_app/lib/domain/value_objects/answer_choice.dart
import 'package:json_annotation/json_annotation.dart';

enum AnswerChoice {
  @JsonValue('STRONGLY_A')
  stronglyA(2),

  @JsonValue('SLIGHTLY_A')
  slightlyA(1),

  @JsonValue('SLIGHTLY_B')
  slightlyB(-1),

  @JsonValue('STRONGLY_B')
  stronglyB(-2);

  const AnswerChoice(this.score);

  final int score;
}
